async-stream-impl/src/lib.rs

```diff
@@ -98,6 +98,42 @@ impl VisitMut for Scrub {
                 syn::visit_mut::visit_expr_mut(self, i);
                 self.is_xforming = prev;
             }
+            syn::Expr::ForLoop(expr) => {
+                syn::visit_mut::visit_expr_for_loop_mut(self, expr);
+                // TODO: Should we allow other attributes?
```

Comment:
What about just `#[await]`?

Also, since this is a macro anyway, we could potentially get more aggressive and transform:

```rust
for await foo in my_stream {
}
```

Reply:
> What about just `#[await]`?

Sounds good to me.

> ```rust
> for await foo in my_stream {
> }
> ```

I think syn doesn't support this syntax yet.

Reply:
(By the way, that TODO comment is about the current implementation confirming that the number of attributes is exactly one.)

Reply:
`syn` doesn't support it... but it may not be too hard to impl. You would have to manually walk the `TokenStream` and find `for` followed by `await`, transform that to `#[await] for` then run ^^.

That can be follow up work though...

Reply:
Done in https://github.com/tokio-rs/async-stream/pull/2/commits/c1d4aefb3c75b1df610348e110d181e4f22a6e07